go/private/rules/cgo.bzl

```diff
@@ -149,6 +149,7 @@ def _cgo_codegen_impl(ctx):
       ),
       OutputGroupInfo(
           go_files = to_set(go_outs),
+          input_go_files = to_set(source.go),
```

Comment:
Is it sufficient to return these? We should hide cgo sources from the compiler in pure mode, even if there's no build tag. "go build" seems to do this.

The best place is probably in the compile builder. We do tag filtering there, and we inspect the imports anyway for strict dependency checking.

Reply:
In here this is all we can do, yes if we need to do more filtering it has to go in the compile wrapper, we should be setting the cgo bool correctly, I was presuming that was sufficient already.